### ISSUE TITLE
Compile-AppInBcContainer should also copy Business Foundation when CopySymbolsFromContainer is used

### ISSUE THREAD

**Issue by aablasco** [microsoft/navcontainerhelper#3563]
**Describe the issue**
We have an app that extends Nos. Series page and table, we use the -CopySymbolsFromContainer in the Compile-AppInBcContainer and it gives a compilation error because of the Business Foundation symbols are missing.

**Scripts used to create container and cause the issue**
Any version 24.x Business Central Container and the HelloWorld app with the following page extension:

```al
namespace DefaultPublisher.noseriestest;
using Microsoft.Foundation.NoSeries;

pageextension 50101 GsiNoSeries extends "No. Series"
{
    trigger OnOpenPage();
    begin
        Message('App published: Hello world');
    end;
}
```

**Full output of scripts**
```
Compile-AppInBcContainer -containerName bcp -credential $gregal -appProjectFolder C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\ -CopySymbolsFromContainer
Using Symbols Folder: C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\.alpackages
Copying System.app from Container
Copying Microsoft_Application.app from Container
Copying Microsoft_Base Application.app from Container
Copying Microsoft_System Application.app from Container
Processing dependency Microsoft_Application_24.0.0.0 ()
Dependency App exists
Processing dependency Microsoft_System_1.0.0.0 ()
Dependency App exists
Compiling...
.\alc.exe /project:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest" /packagecachepath:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\.alpackages" /out:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\output\Gregal_NoSeriesTest_1.0.0.0.app" /BuildBy:"BcContainerHelper,6.0.18" /assemblyprobingpaths:"C:\Program Files\dotnet\shared","C:\ProgramData\BcContainerHelper\Extensions\bcp\.netPackages\Service"
Microsoft (R) AL Compiler version 13.0.14.57639
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'NoSeriesTest' containing '1' files at '14:21:52.170'.

error AL1022: A package with publisher 'Microsoft', name 'Business Foundation', and a version compatible with '24.0.0.0' could not be found in the package cache folders: C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\.alpackages
```

**Additional context**

If we don't use the  -CopySymbolsFromContainer, the apps compiles, which it is actually a workaround for this issue.

```
Compile-AppInBcContainer -containerName bcp -credential $gregal -appProjectFolder C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest
Using Symbols Folder: C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\.alpackages
Processing dependency Microsoft_Application_24.0.0.0 ()
Downloading symbols: Microsoft_Application_24.0.16410.18056.app
Url : http://172.23.118.172:7049/BC/dev/packages?publisher=Microsoft&appName=Application&versionText=24.0.0.0&tenant=default
Adding dependency to System Application from Microsoft
Adding dependency to Business Foundation from Microsoft
Adding dependency to Base Application from Microsoft
Processing dependency Microsoft_System_1.0.0.0 ()
Downloading symbols: Microsoft_System_24.0.18037.0.app
Url : http://172.23.118.172:7049/BC/dev/packages?publisher=Microsoft&appName=System&versionText=1.0.0.0&tenant=default
Processing dependency Microsoft_System Application_24.0.0.0 (63ca2fa4-4f03-4f2b-a480-172fef340d3f)
Downloading symbols: Microsoft_System Application_24.0.16410.18056.app
Url : http://172.23.118.172:7049/BC/dev/packages?appId=63ca2fa4-4f03-4f2b-a480-172fef340d3f&versionText=24.0.0.0&tenant=default
Processing dependency Microsoft_Business Foundation_24.0.0.0 (f3552374-a1f2-4356-848e-196002525837)
Downloading symbols: Microsoft_Business Foundation_24.0.16410.18056.app
Url : http://172.23.118.172:7049/BC/dev/packages?appId=f3552374-a1f2-4356-848e-196002525837&versionText=24.0.0.0&tenant=default
Processing dependency Microsoft_Base Application_24.0.0.0 (437dbf0e-84ff-417a-965d-ed2bb9650972)
Downloading symbols: Microsoft_Base Application_24.0.16410.18056.app
Url : http://172.23.118.172:7049/BC/dev/packages?appId=437dbf0e-84ff-417a-965d-ed2bb9650972&versionText=24.0.0.0&tenant=default
Compiling...
.\alc.exe /project:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest" /packagecachepath:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\.alpackages" /out:"C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\output\Gregal_NoSeriesTest_1.0.0.0.app" /BuildBy:"BcContainerHelper,6.0.18" /assemblyprobingpaths:"C:\Program Files\dotnet\shared","C:\ProgramData\BcContainerHelper\Extensions\bcp\.netPackages\Service"
Microsoft (R) AL Compiler version 13.0.14.57639
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'NoSeriesTest' containing '1' files at '14:33:39.010'.


Compilation ended at '14:33:43.045'.

C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\output\Gregal_NoSeriesTest_1.0.0.0.app successfully created in 28 seconds
C:\ProgramData\BcContainerHelper\Extensions\bcp\my\NoSeriesTest\output\Gregal_NoSeriesTest_1.0.0.0.app
```

I think adding a line in the paths array to copy the Business Foundation app in the AppHandling/Compile-AppInNavContainer.ps1 will solve this issue:

```powershell
    if ($CopySymbolsFromContainer) {
        Invoke-ScriptInBcContainer -containerName $containerName -scriptblock { Param($appSymbolsFolder)
            if (Test-Path "C:\Extensions\*.app") {
                $paths = @(
                    "C:\Program Files\Microsoft Dynamics NAV\*\AL Development Environment\System.app"
                    "C:\Extensions\*.app"
                )
            }
            else {
                $paths = @(
                    "C:\Program Files\Microsoft Dynamics NAV\*\AL Development Environment\System.app"
                    "C:\Applications.*\Microsoft_Application_*.app,C:\Applications\Application\Source\Microsoft_Application.app"
                    "C:\Applications.*\Microsoft_Base Application_*.app,C:\Applications\BaseApp\Source\Microsoft_Base Application.app"
                    "C:\Applications.*\Microsoft_System Application_*.app,C:\Applications\System Application\source\Microsoft_System Application.app",
                    "C:\Applications.*\Microsoft_Business Foundation_*.app,C:\Applications\System Application\source\Microsoft_Business Foundation.app"
                )
            }
```

If the path does not exist because is a previous version, that line is just ignored

**Comment by freddydk** [comment]
Just fixed this this morning - should be in the latest preview.

**Comment by aablasco** [comment]
Yes, it is fixed in the version 6.0.19-preview1211
Thank you for the quick answer
